python/pyspark/sql/functions.py

```diff
@@ -2214,6 +2216,37 @@ def pandas_udf(f=None, returnType=None, functionType=None):
 
        .. seealso:: :meth:`pyspark.sql.GroupedData.apply`
 
+    3. GROUP_AGG
+
+       A group aggregate UDF defines a transformation: One or more `pandas.Series` -> A scalar
+       The returnType should be a primitive data type, e.g, `DoubleType()`.
+       The returned scalar can be either a python primitive type, e.g., `int` or `float`
+       or a numpy data type, e.g., `numpy.int64` or `numpy.float64`.
+
+       StructType and ArrayType are currently not supported.
+
+       Group aggregate UDFs are used with :meth:`pyspark.sql.GroupedData.agg`
+
+       >>> from pyspark.sql.functions import pandas_udf, PandasUDFType
+       >>> df = spark.createDataFrame(
+       ...     [(1, 1.0), (1, 2.0), (2, 3.0), (2, 5.0), (2, 10.0)],
+       ...     ("id", "v"))
+       >>> @pandas_udf("double", PandasUDFType.GROUP_AGG)
+       ... def mean_udf(v):
```

Comment:
shall we include grouping columns?

Reply:
Sorry @cloud-fan, I don't understand this comment, could you elaborate?

Reply:
similar to the `GROUP_MAP` UDF, shall we also consider putting the grouping columns in the parameters of `GROUP_AGG` UDF?

Reply:
IIUC, this is similar to SQL's aggregation, and this aggregation UDF should only take the column for aggregation without grouping columns.

Reply:
ah makes sense

Reply:
I agree with @viirya